core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/searchfor/MetaClassesParticipantsItemProvider.java

```diff
@@ -90,16 +91,15 @@ public boolean isShowSemantics() {
   public void setShowSemantics(boolean showSemantics) {
     this.showSemantics = showSemantics;
   }
-
-  protected boolean isFiltered(Object cls) {
-    if(!showSemantics && !showAbstract)
-      return true;
+  
+  // based on abstract/semantic checks, check if the element is displayed or not
+  protected boolean isDisplayed(Object cls) {

```

Comment:
This should take into account 2 filters at the same time. In this state, a semantic meta-class is displayed when the Semantic filter is checked regardless of the Abstract filter being checked or not. Similarly, an abstract class is displayed when the Abstract filter is checked regardless of the Semantic filter being checked or not.